catalogs/catalog-lakehouse-paimon/src/main/java/com/datastrato/gravitino/catalog/lakehouse/paimon/utils/CatalogUtils.java

```diff
@@ -46,17 +56,55 @@ public static Catalog loadCatalogBackend(PaimonConfig paimonConfig) {
     String metastore = paimonConfig.get(CATALOG_BACKEND);
     Preconditions.checkArgument(
         StringUtils.isNotBlank(metastore), "Paimon Catalog metastore can not be null or empty.");
+
     String warehouse = paimonConfig.get(CATALOG_WAREHOUSE);
     Preconditions.checkArgument(
         StringUtils.isNotBlank(warehouse), "Paimon Catalog warehouse can not be null or empty.");
+
     if (!PaimonCatalogBackend.FILESYSTEM.name().equalsIgnoreCase(metastore)) {
       String uri = paimonConfig.get(CATALOG_URI);
       Preconditions.checkArgument(
-          StringUtils.isNotBlank(uri),
-          String.format("Paimon Catalog uri can not be null or empty for %s.", metastore));
+          StringUtils.isNotBlank(uri), "Paimon Catalog uri can not be null or empty.");
     }
+
+    Map<String, String> allConfig = paimonConfig.getAllConfig();
+    Configuration configuration = new Configuration();
+    allConfig.forEach(configuration::set);
+
     CatalogContext catalogContext =
-        CatalogContext.create(Options.fromMap(paimonConfig.getAllConfig()));
-    return CatalogFactory.createCatalog(catalogContext);
+        CatalogContext.create(Options.fromMap(paimonConfig.getAllConfig()), configuration);
+
+    AuthenticationConfig authenticationConfig = new AuthenticationConfig(allConfig);
+    if (authenticationConfig.isSimpleAuth()) {
+      return CatalogFactory.createCatalog(catalogContext);
+    } else if (authenticationConfig.isKerberosAuth()) {
+      configuration.set(HADOOP_SECURITY_AUTHORIZATION, "true");
+      configuration.set(HADOOP_SECURITY_AUTHENTICATION, "kerberos");
+
+      switch (PaimonCatalogBackend.valueOf(metastore.toUpperCase(Locale.ROOT))) {
+        case FILESYSTEM:
+          initKerberosAndReturnRealm(allConfig, configuration);
```

Comment:
could you explain how this effects Paimon catalog, since `CatalogFactory.createCatalog(catalogContext)` is isolated.

Reply:
`CatalogFactory.createCatalog(catalogContext)` will create Filesystem client, and after `loginUserFromKeytab`, 
the client can get the current ugi to initialize itselt.

Reply:
Where does the ugi  saved? thread local variable? should we remove it ?

Reply:
@yuqi1129  WDYT?

Reply:
`initKerberosAndReturnRealm` will initialize the Kerberos user and make the current user is the one that has just logged in, then the default user used in the later steps is that one.  

> Where does the ugi saved? thread local variable? should we remove it ?

It saves local variations in the thread. It's just initialized in the catalog create procedure, so it's okay to keep it as is.

Reply:
sounds like it saves the user in a thread-local variable,  could the implement grant the correctness of the user in multi user environment? If yes,  how?

Reply:
> could the implement grant the correctness of the user in multi user environment? If yes, how?

There are no multiple users for a catalog as it's passed from the catalog creation process. What you mean multiple users may refer to users who were previously passed from the SDK API.

Reply:
@FANNG1 After carefully check, this part is okay with me.

Reply:
@yuqi1129 could you explain the reason?

Reply:
Each catalog will have a separate class loader and a single Kerberos login user, so it's okay to do so.